### PR TITLE
Proxy txn cleanup

### PR DESCRIPTION
**Pull request by a-canary**
Work-In-Progress
I'm using the CI to verify multiple small changes before squashing.

### CODE REVIEW

File: proxy/http/Http1Transaction.h [hunk 1]
```diff
@@ -37,6 +37,7 @@ class Http1Transaction : public ProxyTransaction
 
   ////////////////////
   // Methods
+  void init_transaction(Http1ClientSession *ssn); // overload, not override
```
Comment (maskit) [inline proxy/http/Http1Transaction.h]: I'm fine with overloading, but why do we want this method? I think we can pass Http(1|2|3)ClientSesion to constructors.

File: proxy/http/Http1Transaction.h [hunk 2]
```diff
@@ -26,110 +26,43 @@
...
-
-  void
-  set_reader(IOBufferReader *reader)
-  {
-    sm_reader = reader;
-  }
 
+  ////////////////////
+  // Methods
+  void init_transaction(Http1ClientSession *ssn); // overload, not override
```
Comment (maskit) [inline proxy/http/Http1Transaction.h]: I'm fine with overloading, but why do we want this method? I think we can pass Http(1|2|3)ClientSesion to constructors.

Comment (maskit) [reply proxy/http/Http1Transaction.h]: Ah, I read the commit message. You're going to make constructors. +1.

**Comment by SolidWallOfCode** [comment]
Shouldn't this have been a draft PR?

**Comment by a-canary** [comment]
[approve ci]

**Comment by a-canary** [comment]
[approve ci autest]

**Comment by a-canary** [comment]
[approve ci autest]

**Comment by a-canary** [comment]
[approve ci autest]

**Comment by dragon512** [comment]
[approve ci]

**Comment by maskit** [comment]
FYI: Naming conventions
https://cwiki.apache.org/confluence/display/TS/Coding+Style

**Pull request closed without merging**